drivers/timer/nrf_rtc_timer.c

```diff
@@ -117,21 +130,53 @@ void z_clock_set_timeout(s32_t ticks, bool idle)
 	ticks = (ticks == K_FOREVER) ? MAX_TICKS : ticks;
 	ticks = max(min(ticks - 1, (s32_t)MAX_TICKS), 0);
 
-	k_spinlock_key_t key = k_spin_lock(&lock);
-	u32_t cyc, t = counter();
-
-	/* Round up to next tick boundary */
-	cyc = ticks * CYC_PER_TICK + counter_sub(t, last_count);
-	cyc += (CYC_PER_TICK - 1);
-	cyc = (cyc / CYC_PER_TICK) * CYC_PER_TICK;
-	cyc += last_count;
-
-	if (counter_sub(cyc, t) < MIN_DELAY) {
-		cyc += CYC_PER_TICK;
+	/*
+	 * Get the requested delay in tick-aligned cycles.  Increase
+	 * by one tick to round up so we don't timeout early due to
+	 * cycles elapsed since the last tick.  Cap at the maximum
+	 * tick-aligned delta.
+	 */
+	u32_t cyc = min((1 + ticks) * CYC_PER_TICK, MAX_DELAY);
```

Comment:
Doesn't it delay all announcements?
Is that workaround for rounding down in ms to tick conversion?

Reply:
No.  It's a workaround for having to adjust for a non-zero time-since-last-tick in the locked region.  The only time it adds an "unnecessary" tick is if you get to this point in the code less than 30.6 us after RTC counter produced a compare event and the timer ISR completed update of `last_count`.  That's not going to happen very often.

Reply:
OK